orchestrator/gravity_utils/src/types/ethereum_events.rs

```diff
@@ -41,20 +41,20 @@ pub struct ValsetUpdatedEvent {
 struct ValsetDataBytes {
     pub event_nonce: u64,
     pub reward_amount: Uint256,
-    pub reward_token: Option<EthAddress>,
+    pub reward_denom: String,
+    pub reward_recipient: String,
     pub members: Vec<ValsetMember>,
 }
 
+// TODO refactor the entire file to parse by arg index instead of code duplication.
```

Comment:
Could you do this now? This code is getting error prone. Also, I think you should be able to take the bytes of the address directly instead of going through utf8 strings and stuff between the Go and Rust sides.

Reply:
I've left that comment and created a new GitHub issue,  in order to rewrite that file in a separate PR and commit to paying more attention to the detail. Alos nut just Valset event function must be rewritten, but all the functions here, so it's out of scope for that particular PR.
Here is that issue: https://github.com/onomyprotocol/arc/issues/93